src/votekit/election_types.py

```diff
@@ -1,98 +1,125 @@
-        self.elected: set = set()
-        self.eliminated: set = set()
-        self.seats = seats
-        self.threshold = self.get_threshold()
+        self.transfer: Callable = transfer
+        self.seats: int = seats
+
+        fp_votes = compute_votes(profile.get_candidates(), profile.get_ballots())
+        fp_order = [
+            y[0] for y in sorted(fp_votes.items(), key=lambda x: x[1], reverse=True)
+        ]
+
+        self.election_state: ElectionState = ElectionState(
+            curr_round=0, elected=[], eliminated=[], remaining=fp_order, profile=profile
+        )
+        self.threshold: float = self.get_threshold()
 
     # can cache since it will not change throughout rounds
     def get_threshold(self) -> int:
         """
-        Droop qouta
+        Droop quota
         """
-        return int(self.profile.num_ballots() / (self.seats + 1) + 1)
+        return int(self.election_state.profile.num_ballots() / (self.seats + 1) + 1)
 
     def next_round(self) -> bool:
         """
         Determines if the number of seats has been met to call election
         """
-        return len(self.elected) != self.seats
+        return len(self.election_state.get_all_winners()) != self.seats
 
-    def run_step(self, profile: PreferenceProfile) -> tuple[PreferenceProfile, Outcome]:
+    def run_step(self) -> ElectionState:
         """
         Simulates one round an STV election
         """
-        candidates: list = profile.get_candidates()
-        ballots: list = profile.get_ballots()
-        fp_votes: dict = compute_votes(candidates, ballots)
-
-        # if number of remaining candidates equals number of remaining seats
-        if len(candidates) == self.seats - len(self.elected):
-            # TODO: sort remaing candidates by vote share
-            self.elected.update(set(candidates))
-            return profile, Outcome(
-                elected=self.elected,
-                eliminated=self.eliminated,
-                remaining=set(candidates),
-                votes=fp_votes,
-            )
-
-        for candidate in candidates:
-            if fp_votes[candidate] >= self.threshold:
-                self.elected.add(candidate)
-                candidates.remove(candidate)
-                ballots = self.transfer(candidate, ballots, fp_votes, self.threshold)
-
-        if self.next_round():
+        ##TODO:must change the way we pass winner_votes
+        remaining: list = self.election_state.remaining
+        ballots: list = self.election_state.profile.get_ballots()
+        fp_votes: dict = compute_votes(remaining, ballots)
+        fp_order = [
+            y[0] for y in sorted(fp_votes.items(), key=lambda x: x[1], reverse=True)
+        ]
+        elected = []
+        eliminated = []
+
+        # if number of remaining candidates equals number of remaining seats, everyone is elected
+        if len(remaining) == self.seats - len(self.election_state.get_all_winners()):
+            elected = fp_order
```

Comment:
we should set self.election_state and return here because the election is over is this conditional is true